### DISCUSSION TITLE
Project blinking and crashing when exporting to android

### DISCUSSION THREAD

**Discussion by InfiniteNex** [GodotNuts/GodotFirebase#188] [Q&A] [answered]
I am building a mobile game using Firebase. The project works flawlessly when testing on PC, but when exporting to my phone, either by .apk or by usb debugging, the game screen flickers and closes itself.

I tried a lot of things, and am 100% sure its the Firebase module.
I've installed it in the suggested way, by grabbing the repository's latest version.

For a minimum testing project I created the most simple of scripts:
a button and a label
```javascript
extends Control

#the following are X-ed out but am using real ones in the project
var username = "xxxxxxx"
var email = "xxx@xx.xx"
var password = "xxxxxxxx"

func _ready():
	Firebase.Auth.connect("login_succeeded", self, "_on_login_successful")

func _on_Button_pressed():
	Firebase.Auth.login_with_email_and_password(email, password)

func _on_login_successful(auth):
	print("Login successful")
	$label.text = "Login successful"

func _on_login_failed(code, message):
	$label.text = ("Login failed. Reason: %s") %message

```

On PC the code above runs without a problem, but on my phone it flickers and closes.
I cant figure any further possible solutions and need some help with this.

**Comment by BearDooks** [comment]
@InfiniteNex have you been able to use the ADB to check what might be going on?

It has been a while since I had to use it, but I ended up having an issue that I was able to track down pretty quickly with it.

~ Chuck

**Reply by InfiniteNex** [reply]
No. All I have used so far is the debug deploy through USB. Do you have a guide on how to use the ADB?

**Reply by BearDooks** [reply]
@InfiniteNex it has been a while but this sums it up pretty well.

https://godotengine.org/qa/83771/how-to-know-reason-why-the-game-crashed-on-android-device

When I was running it on my machine for debugging I had to use the command `adb logcat *:S Godot:D` . Not sure if that still works, but it should point you in the right direction

~ Chuck

**Reply by InfiniteNex** [reply]
ADB logcat gives me errors for a bunch of Firebase classes not found and couldnt be loaded. The above is just one of the many.
I copied part of it in a comment below

**Comment by WolfgangSenff** [comment]
I know this isn't very helpful, but I've heard recently there's a lot of bugs right now with Android exports specifically. I can try to attempt to figure out what's going on myself too at the same time tonight after work. Thanks for pointing it out - if it is one with our stuff, I'll go ahead and open a ticket for it.

**Reply by InfiniteNex** [reply]
After running adb logcat i got a bunch of errors of this type:

**ERROR**: Method failed. Returning: ERR_PARSE_ERROR
05-10 19:21:57.061 14276 14313 E godot   :    At: modules/gdscript/gdscript.cpp:801:load_byte_code() - Method failed. Returning: ERR_PARSE_ERROR
05-10 19:21:57.061 14276 14313 E godot   : **ERROR**: Cannot load byte code from file 'res://addons/godot-firebase/database/database.gdc'.
05-10 19:21:57.061 14276 14313 E godot   :    At: modules/gdscript/gdscript.cpp:2282:load() - Condition "err != OK" is true. Returned: RES()
05-10 19:21:57.061 14276 14313 E godot   : **ERROR**: Failed loading resource: res://addons/godot-firebase/database/database.gdc. Make sure resources have been imported by opening the project in the editor at least once.
05-10 19:21:57.061 14276 14313 E godot   :    At: core/io/resource_loader.cpp:279:_load() - Condition "found" is true. Returned: RES()
05-10 19:21:57.062 14276 14313 I godot   : Loading resource: res://addons/godot-firebase/storage/storage.gdc
05-10 19:21:57.074 14276 14313 I godot   : Loading resource: res://addons/godot-firebase/dynamiclinks/dynamiclinks.gdc
05-10 19:21:57.078 14276 14313 E godot   : **SCRIPT ERROR**: Parse Error: The class "FirebaseAuth" was found in global scope, but its script couldn't be loaded.
05-10 19:21:57.078 14276 14313 E godot   :    At: res://addons/godot-firebase/firebase/firebase.gdc:19:GDScript::load_byte_code() - Parse Error: The class "FirebaseAuth" was found in global scope, but its script couldn't be loaded.


A bunch of Firebase classes not found and couldnt be loaded. The above is just one of the many.

**Reply by BearDooks** [reply]
@InfiniteNex thanks for this. Well I can clearly see there is something going on with our plugin. 

If you give me some time, I can make a quick test project and export it and see if I am seeing the same thing, or if there is something else we need to do.

Thanks

~ Chuck

**Comment by WolfgangSenff** [comment]
I've run into this issue before. It has something to do with initialization and actually opening the project. @InfiniteNex - do me a favor: next time you have a bit of time to test, try opening the actual Godot Firebase project file that's included in the plugin repository, then go back to your project, open the plugins, and refresh the plugins. Make sure Firebase is turned on there as well - it should be by default, but just in case. Unfortunately I'm not remembering exactly what I did to fix it, but I feel like something along those lines did it.

**Reply by InfiniteNex** [reply]
Well I just tried what you said. I opened my project and within it opened Firebase.tscn, and turned off and back on again all the plugins. No difference in the result.
I hope I actually understood you correctly.

**Comment by InfiniteNex** [comment, marked as answer]
THE ISSIUE HAS BEEN FIXED! I randomly turned off the "Custom Docs" plugin, and now everything works as intended. Its a super non-obvious fix but it worked!

Thanks to everyone who tried to help me out.

**Reply by BearDooks** [reply]
@InfiniteNex thanks for letting us know what it was. I will bring that back to the team
